### PR TITLE
feat: Kyber reserve discovery using getTradingReserves

### PR DESCRIPTION
**Pull request by dekz**
## Description

tl;dr We had a whitelist approach, reserveIds are almost impossible to find (documented), new ones seem to have been added. We now look for up to 6 kyber reserves, skipping any Bridged reserves (`0xbb`) in the Sampler, double checking in `sampler_operations`.

Refactor sampler to allow some per source failures without taking down the entire sampler result.

Added Mooniswap V2.1

### Simbots
[Kyber only](https://metabase.spaceship.0x.org/dashboard/139?run_id=kyber-4-offsets-5&adjusted_amount_bought_win_tolerance__bps_=0.5)
[Global](https://metabase.spaceship.0x.org/dashboard/139?run_id=kyber-mooni-21&adjusted_amount_bought_win_tolerance__bps_=0.5)
[Global - buys](https://metabase.spaceship.0x.org/dashboard/139?run_id=kyber-mooni-21&adjusted_amount_bought_win_tolerance__bps_=0.5)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Review by mzhu25 — APPROVED** [review]
lgtm, just a quick q
